jvb/src/main/java/org/jitsi/videobridge/metrics/MetricsContainer.java

```diff
@@ -0,0 +1,171 @@
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.metrics;
+
+import io.prometheus.client.*;
+import io.prometheus.client.exporter.common.*;
+import org.jitsi.videobridge.*;
+import org.json.simple.*;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * {@code MetricsContainer} gathers and exports metrics from a {@link Videobridge} instance.
+ */
+public final class MetricsContainer
+{
+    /**
+     * The singleton instance of {@code MetricsContainer}.
+     */
+    private static final MetricsContainer INSTANCE = new MetricsContainer();
+
+    /**
+     * Namespace prefix added to all metrics.
+     */
+    public static final String METRICS_NAMESPACE = "jitsi";
+
+    /**
+     * Metrics are just wrappers for different Prometheus metric types.
+     * These are wrapped {@link Gauge Gauges} that also provide atomic operations.
+     */
+    private final Map<String, Metric> metrics;
+
+    /**
+     * Initializes a new {@code MetricsContainer} instance.
+     */
+    private MetricsContainer()
+    {
+        metrics = new HashMap<>();
+    }
+
+    /**
+     * Returns the singleton instance of {@code MetricsContainer}.
+     *
+     * @return the singleton instance of {@code MetricsContainer}.
+     */
+    public static MetricsContainer getInstance()
+    {
+        return INSTANCE;
+    }
+
+    /**
+     * Returns the metrics in this instance as a JSON string.
+     *
+     * @return a JSON string of the metrics in this instance
+     */
+    public String getJsonString()
+    {
+        Map<String, Object> map = new HashMap<>(metrics.size());
+        metrics.forEach((key, metric) -> map.put(key, metric.getMetricValue()));
+        return new JSONObject(map).toJSONString();
+    }
+
+    /**
+     * Returns the metrics in this instance in the Prometheus text-based format.
+     *
+     * @param contentType the Content-Type header string
+     * @return a JSON string of the metrics in this instance
+     */
+    public String getPrometheusMetrics(String contentType)
+    {
+        StringWriter writer = new StringWriter();
+        try
+        {
+            TextFormat.writeFormat(contentType, writer, CollectorRegistry.defaultRegistry.metricFamilySamples());
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
+        return writer.toString();
+    }
+
+    /**
+     * Creates and registers a {@link BooleanMetric} with the given name and help string.
+     *
+     * @param name the name of this metric
+     * @param help the descriptive help string of this metric
+     * @return the registered metric
+     */
+    public BooleanMetric registerBooleanMetric(String name, String help)
+    {
+        BooleanMetric booleanGauge = null;
+        if (!metrics.containsKey(name))
+        {
+            booleanGauge = new BooleanMetric(name, help, METRICS_NAMESPACE);
+            metrics.put(name, booleanGauge);
+        }
+        return booleanGauge;
```

Comment:
Did you mean to return the existing metric if a metric with that name was already registered? Or just return `null` as a failure indication?

I suggest logging a warning or even throwing an exception, because I don't anticipate this happening legitimately.

The same applies to the other `register` methods.

Reply:
I wasn't sure whether I should throw an exception or return an existing one so I left it as it is for now.
I believe the test failures are due to this.
As an example: in [ConferenceTest.kt](https://github.com/jitsi/jitsi-videobridge/blob/master/jvb/src/test/kotlin/org/jitsi/videobridge/ConferenceTest.kt#L31), initializing new `Videobridge.Statistics()` results in null metrics. This leads to `NullPointerException` when calling methods such as `totalConferencesCreated.incrementAndGetLong()`.

Should we just return the metric if it exists (and perhaps log a warning)?